lib/connectors/base/connector.rb

```diff
@@ -65,7 +65,7 @@ def initialize(configuration: {}, job_description: nil)
         error_monitor = Utility::ErrorMonitor.new
         @tolerable_error_helper = Connectors::TolerableErrorHelper.new(error_monitor)
 
-        @configuration = configuration.dup || {}
+        @configuration = job_description&.configuration&.dup || configuration&.dup || {}
```

Comment:
`|| configuration&.dup`
This fallback is for backwards compatibility?

Reply:
No. Heartbeat is also in connector implementation, and it doesn't tie to a particular job.

Reply:
We need to untie heartbeat from actual execution better in future IMO, but can't think of a simple way to do so yet

Reply:
Agree. Probably have a separate class called `Heartbeat`, so the `Connector` is only responsible for syncing data.